### PR TITLE
update docker process

### PR DESCRIPTION
**Pull request by mpmdean**
This implements some changes related to differences in how we configure docker

* It updates the docs to use docker-compose and to account for the changes in username within the container. 
* It also tries to push docker images with GH actions

### CODE REVIEW

**Comment by mpmdean** [comment]
@lajerm would you be able to check this PR?

File: docs/source/user/usedocker.rst [hunk 1]
```diff
@@ -7,66 +7,47 @@ edrixs and docker
...
+          image: edrixs/edrixs
+          volumes:
+            - ./:/home/rixs
+          working_dir: /home/rixs
 
-  it will take a while to pull the image from `Docker Hub <https://cloud.docker.com/repository/docker/edrixs/edrixs/>`_ for the first time, while, it will launch the local one very fast at the next time.
+  and execute ::
 
-  * ``-u rixs`` means using a default user ``rixs`` to login the Ubuntu Linux, the password is ``rixs``.
+    docker-compose up
```
Comment (lajerm) [inline docs/source/user/usedocker.rst:36]: docker compose up

File: .github/workflows/docker-image-publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Publish Docker image
+
+on:
```
Comment (mrakitin) [inline .github/workflows/docker-image-publish.yml:3]:
```suggestion
on:
  workflow_dispatch:
```

**Review by mrakitin — APPROVED** [review]
Looks good to me!

**Comment by mpmdean** [comment]
@mrakitin 

The GH action works but it tags the docker image as "master", presumably due to the form of this line
https://github.com/NSLS-II/edrixs/blob/97f6fdee36ff56ba45c0b7ef1613977d7c35808f/.github/workflows/docker-image-publish.yml#L34

This line in the instructions defaults to the "latest tag"
https://github.com/NSLS-II/edrixs/blob/97f6fdee36ff56ba45c0b7ef1613977d7c35808f/docs/source/user/usedocker.rst?plain=1#L21

I would lead towards changing the label here? What do you think?

**Pull request merged**